src/main/java/teammates/storage/sqlapi/AccountRequestDb.java

```diff
@@ -0,0 +1,140 @@
+package teammates.storage.sqlapi;
+
+import java.time.Instant;
+import java.util.List;
+
+import org.hibernate.Session;
+
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.HibernateUtil;
+import teammates.storage.sqlentity.AccountRequest;
+
+import jakarta.persistence.TypedQuery;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
+
+/**
+ * Generates CRUD operations for AccountRequest.
+ *
+ * @see AccountRequest
+ */
+public final class AccountRequestDb extends EntitiesDb<AccountRequest> {
+    private static final AccountRequestDb instance = new AccountRequestDb();
+
+    private AccountRequestDb() {
+        // prevent instantiation
+    }
+
+    public static AccountRequestDb inst() {
+        return instance;
+    }
+
+    /**
+     * Creates an AccountRequest in the database.
+     */
+    public AccountRequest createAccountRequest(AccountRequest accountRequest)
+            throws InvalidParametersException, EntityAlreadyExistsException {
+        assert accountRequest != null;
+
+        accountRequest.sanitizeForSaving();
+        if (!accountRequest.isValid()) {
+            throw new InvalidParametersException(accountRequest.getInvalidityInfo());
+        }
+
+        // don't need to check registrationKey for uniqueness since it is generated using email + institute
+        if (getAccountRequest(accountRequest.getEmail(), accountRequest.getInstitute()) != null) {
+            throw new EntityAlreadyExistsException(
+                String.format(ERROR_CREATE_ENTITY_ALREADY_EXISTS, accountRequest.toString()));
+        }
+
+        persist(accountRequest);
+        return accountRequest;
+    }
+
+    /**
+     * Get AccountRequest by {@code accountRequestId} from database.
+     */
+    public AccountRequest getAccountRequest(int accountRequestId) {
```

Comment:
We may not need this now, but will change to mainly use id once everything has been migrated.

Reply:
I have removed this method for now.